modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/SearchAsYouTypeFieldMapper.java

```diff
@@ -755,6 +755,11 @@ public Iterator<Mapper> iterator() {
         return concat;
     }
 
+    @Override
+    public Iterator<Mapper> sourcePathUsedBy() {
+        return iterator();
```

Comment:
Do we actually need this override?  It's the same as the superclass.

Reply:
It's not using the superclass's, it's using its own (`SearchAsYouTypeFieldMapper.iterator()`), which overrides the superclass's.

Reply:
Sorry, you meant that I could use the super's `iterator()`, which means that it is essentially the same as the super's implementation for `sourcePathUsedBy()`- not only this is true, it means that this is a bug!
I wanted to make it explicit that there is a default behavior for `sourcePathUsedBy()` and a specific override for it, but still not duplicate the code, but what I essentially did is as if I just used `FieldMapper.iterator()`, which we wanted to avoid 🤦‍♂️ 

I will fix

Reply:
Fixed with the [latest commit](https://github.com/elastic/elasticsearch/pull/97092/commits/4b99fc7f3d30f16568eef8dedfc44401d042a413).
Thanks for pointing it out! 🙏